### PR TITLE
Start task scheduler at end of startup.

### LINKED ISSUE #22648: delay starting of task runner until metabase has initialized
Metabase is inoperable until initialization completes. Part of this initialization is to start our task runner based on quartz. 

The scheduler is started and then all tasks are initialized. The downside of this is that if there are a large number of databases (500 or so) there are an average of 10 syncs every minute. This causes us to start syncing databases while still trying to initialize the application.

The Quartz library luckily offers a `standby` mode allowing us to do trigger work and schedule things without jobs firing yet. We can then go from standby mode to started when initialization is complete.

### PR DESCRIPTION
**Pull request by dpsutton**
Fixes #22648 

The sync db task init does a lot of db work to be addressed in a later
commit. But it also begins firing tasks _during startup_. Lets prevent
syncs and other tasks while we are starting up until we have finished.

Scheduler can be set to standby mode while we wait on finishing up
initializing and then start after.

### CODE REVIEW

File: src/metabase/core.clj [hunk 1]
```diff
@@ -105,7 +105,7 @@
     (init-status/set-progress! 0.7)
 
     ;; Now start the task runner
```
Comment (howonlee) [inline src/metabase/core.clj:107]: nit: change seems to invalidate comment here, inasmuch as 'start' vs 'init' is kinda hard to differentiate

**Review by howonlee — APPROVED** [review]
I understand that the atom is not actually used ... atomically ... in the init / start thing but might be good to like, make explicit this decision, assuming it was a decision

also this bit just doesn't seem to have a test - but if we're driving down a turbo-mega-brobdingnagian startup time then we gotta have some kinda profiling in the CI which would seem to require a test?

**Comment by dpsutton** [comment]
> I understand that the atom is not actually used ... atomically ... in the init / start thing but might be good to like, make explicit this decision, assuming it was a decision

It is used atomically. It is used in `(when (compare-and-set! quartz-scheduler nil new-scheduler)` to ensure that init-tasks! is only called when the scheduler is instantiated to ensure tasks are all initialized against the same schedule and not in a race when another scheduler is set.


> also this bit just doesn't seem to have a test - but if we're driving down a turbo-mega-brobdingnagian startup time then we gotta have some kinda profiling in the CI which would seem to require a test?

That's an interesting idea. I think this is unfortunately difficult to do. The problems we are seeing with startup times only manifest in conditions like 500+ databases and high garbage in a database. This kind of thing makes sense to check periodically or perhaps in a more intensive suite prior to releases. We change the startup so infrequently that running on every commit sounds like very low signal.

**Comment by howonlee** [comment]
> > I understand that the atom is not actually used ... atomically ... in the init / start thing but might be good to like, make explicit this decision, assuming it was a decision
> 
> It is used atomically. It is used in `(when (compare-and-set! quartz-scheduler nil new-scheduler)` to ensure that init-tasks! is only called when the scheduler is instantiated to ensure tasks are all initialized against the same schedule and not in a race when another scheduler is set.

I don't mean with reference to other schedulers, I mean with reference to the same scheduler getting all mutated. Which I feel is legitimately less of a concern but kinda weird to think of it not just being all in one `swap!`

> 
> > also this bit just doesn't seem to have a test - but if we're driving down a turbo-mega-brobdingnagian startup time then we gotta have some kinda profiling in the CI which would seem to require a test?
> 
> That's an interesting idea. I think this is unfortunately difficult to do. The problems we are seeing with startup times only manifest in conditions like 500+ databases and high garbage in a database. This kind of thing makes sense to check periodically or perhaps in a more intensive suite prior to releases. We change the startup so infrequently that running on every commit sounds like very low signal.

(I note, having to go fix the JSON shenanigans myself, lol)

**Pull request merged**